### PR TITLE
Add import/export rake task for ConfigReport

### PR DESCRIPTION
**Pull request by avitova**

### CODE REVIEW

File: lib/tasks/export.rake [hunk 1]
```diff
@@ -0,0 +1,34 @@
+require 'csv'
+
+namespace :reports do
```
Comment (lzap) [inline lib/tasks/export.rake]: Please rename to host_reports.

File: lib/tasks/import.rake [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+      log.id = l['id']
+      log.source_id = l['source_id']
+      log.message_id = l['message_id']
+      log.report_id = l['report_id']
+      log.level_id = l['level_id']
+      User.without_auditing do
+        User.as_anonymous_admin do
+          log.save!
+        end
+      end
```
Comment (lzap) [inline lib/tasks/import.rake:17]: This is a bit slow, try to wrap it into a single transaction:

    Log.transaction do ... end

File: lib/tasks/import.rake [hunk 3]
```diff
@@ -0,0 +1,57 @@
+namespace :reports do
```
Comment (lzap) [inline lib/tasks/import.rake]: Please add rake comment, something like "This task is only for development purposes, database tables must be clean before importing."

File: lib/tasks/export.rake [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+  task :export => :environment do
```
Comment (lzap) [inline lib/tasks/export.rake:5]: Please add rake comment, something like "Export all reports as CSV files. This task is only for development purposes."

File: lib/tasks/import.rake [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+      report.status = JSON.parse(r['status'])
+      report.metrics = JSON.parse(r['metrics'])
+      report.type = r['type']
+      report.origin = r['origin']
+      User.without_auditing do
+        User.as_anonymous_admin do
+          report.save!
+        end
+      end
+    end
```
Comment (lzap) [inline lib/tasks/import.rake:67]: At the end of each file, plese set PG sequence to the highest seen ID + 1: https://www.postgresql.org/docs/9.1/sql-altersequence.html

**Comment by avitova** [comment]
Alter sequence does not allow subqueries, but this should work the same.

File: lib/tasks/import.rake (hunk 2 above)
Comment (lzap) [reply lib/tasks/import.rake:17]: You need to wrap the whole loop in a transaction, that should speed things up.

The same for sources, messages, reports.

File: lib/tasks/import.rake [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+      report.origin = r['origin']
+      Report.transaction do
+        User.without_auditing do
+          User.as_anonymous_admin do
+            report.save!
+          end
+        end
+      end
+    end
+    ActiveRecord::Base.connection.exec_query("SELECT setval('reports_id_seq', (SELECT MAX(id) + 1 FROM reports), false)")
```
Comment (lzap) [inline lib/tasks/import.rake:68]: Great, now please do this for all the tables.

File: lib/tasks/export.rake [hunk 7]
```diff
@@ -0,0 +1,35 @@
+require 'csv'
+
+namespace :foreman_host_reports do
```
Comment (lzap) [inline lib/tasks/export.rake]: Can you rename this to just `host_reports` to match the existing namespace?

File: lib/tasks/import.rake [hunk 8]
```diff
@@ -0,0 +1,70 @@
+namespace :foreman_host_reports do
```
Comment (lzap) [inline lib/tasks/import.rake]: Can you rename this to just `host_reports` to match the existing namespace?

**Review by lzap — CHANGES_REQUESTED** [review]
Other than that, works great as long as the database is cleaned up prior import:

```
[lzap@nuc foreman]$ psql foreman-dev
psql (13.4)
Pro získání nápovědy napište "help".

foreman-dev=# truncate logs, messages, sources, reports;
```

**Comment by avitova** [comment]
Sure, fixed.

**Comment by lzap** [comment]
Thanks for the code, I am closing as we want to re-focus on other things.

**Pull request closed without merging**